test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatWhyTests.cs

```diff
@@ -263,5 +277,247 @@ await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+            var addPackageCommandRunner = new AddPackageReferenceCommandRunner();
+            var addPackageResult = await addPackageCommandRunner.ExecuteCommand(addPackageArgs, new MSBuildAPIUtility(logger));
+
+            var whyCommandArgs = new WhyCommandArgs(
+                    project.ProjectPath,
+                    packageZ.Id,
+                    [projectFramework],
+                    reportRenderer,
+                    logger,
+                    CancellationToken.None);
+
+            // Act
+            var result = await WhyCommandRunner.ExecuteCommand(whyCommandArgs);
+
+            // Assert
+            var output = logger.ShowMessages();
+
+            var expectedOutput = $@"{{
+  ""version"": 1,
+  ""parameters"": ""--framework {projectFramework}"",
+  ""project"": ""{project.ProjectName}"",
+  ""package"": ""{packageZ.Id}"",
+  ""dependencyGraphs"": []
+}}";
+
+            Assert.Equal(ExitCodes.Success, result);
+            Assert.Contains(expectedOutput, output);
+        }
+
+        [Fact]
+        public async Task WhyCommand_JsonRendererDirectPackageFound_SucceedsAsync()
+        {
+            // Arrange
+            var pathContext = new SimpleTestPathContext();
+            var projectFramework = "net472";
+            var project = XPlatTestUtils.CreateProject(ProjectName, pathContext, projectFramework);
+
+            var packageX = XPlatTestUtils.CreatePackage("PackageX", "1.0.0");
+            var packageY = XPlatTestUtils.CreatePackage("PackageY", "1.0.1");
+
+            packageX.Dependencies.Add(packageY);
+
+            project.AddPackageToFramework(projectFramework, packageX);
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                pathContext.PackageSource,
+                PackageSaveMode.Defaultv3,
+                packageX,
+                packageY);
+
+            var logger = new TestCommandOutputLogger(_testOutputHelper);
+            var reportRenderer = new WhyJsonRenderer(logger);
+            var addPackageArgs = XPlatTestUtils.GetPackageReferenceArgs(logger, packageX.Id, packageX.Version, project);
+            var addPackageCommandRunner = new AddPackageReferenceCommandRunner();
+            var addPackageResult = await addPackageCommandRunner.ExecuteCommand(addPackageArgs, new MSBuildAPIUtility(logger));
+
+            var whyCommandArgs = new WhyCommandArgs(
+                    project.ProjectPath,
+                    packageX.Id,
+                    [projectFramework],
+                    reportRenderer,
+                    logger,
+                    CancellationToken.None);
+
+            // Act
+            var result = await WhyCommandRunner.ExecuteCommand(whyCommandArgs);
+
+            // Assert
+            var output = logger.ShowMessages();
+
+            var expectedOutput = $@"{{
+  ""version"": 1,
+  ""parameters"": ""--framework {projectFramework}"",
+  ""project"": ""{project.ProjectName}"",
+  ""package"": ""{packageX.Id}"",
```

Comment:
should the name here be more expliclt, package id? or just "id" to match the rest?

Reply:
The `nuget why` docs use the nomenclature of just `package` so maybe I should rename the `DependencyNodeContract` field from `id` to `Package`? Then it'll be consistent with `WhyJsonReportContract`.

If you prefer `packageId` I think we should also rename `project` to `projectId`.

What do you think?